### PR TITLE
MES-1477: Add journal poller Lambda

### PR DESCRIPTION
**Pull request by johnmorleybjss**
An initial version of the journal poller Lambda.

* Queries active examiners as the basis for other dataset queries
* Runs queries for each dataset and transforms the results into an `ExaminerWorkSchedule` JSON.
* Transfers all of the journals over to DynamoDB

Has some shortcomings to be addressed later on:
* Does not yet compare hashes with what's already in DynamoDB, it simply transfers all journals.
* Window sizes are fixed, these should be configurable.
* Needs some tests adding + refactoring work.
* Query optimisation is required - simply taken the queries as-is from `converted-queries.sql` in this repo.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,64 @@
+{
+  "name": "mes-microservices-starter-template",
+  "version": "0.1.0",
+  "author": "DVSA",
+  "license": "MIT",
+  "description": "A starter template for lambda microservices in the MES project",
+  "homepage": "https://github.com/dvsa/mes-microservices-starter-template#README.md",
+  "repository": {
```
Comment (rossfauldsbjss) [inline package.json]: Need the name,version, description, homepage updating with the name of this service.

File: serverless.yml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    start:
+      port: 8000
+      inMemory: true
+      migrate: true
+  webpack:
+    webpackConfig: 'webpack-sls-offline.config.js'
+
+resources:
+  Resources:
+    penaltyDocuments:
```
Comment (rossfauldsbjss) [inline serverless.yml]: Needs renaming from penaltyDocuments

Comment (ammarhaiderbjss) [reply serverless.yml]: Wonder where that came from 👀

File: src/functions/pollJournals/framework/handler.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+export async function handler(event: APIGatewayProxyEvent, fnCtx: Context): Promise<Response> {
+  const examinerBatchSize = Number.parseInt(process.env.EXAMINER_BATCH_SIZE || '250', 10);
+  const pool: mysql.Pool = mysql.createPool({
+    host: process.env.MYSQL_HOST,
+    database: process.env.MYSQL_DATABASE,
+    user: process.env.MYSQL_USER,
+    password: process.env.MYSQL_PASSWORD,
+    connectionLimit: 50,
+  });
+
```
Comment (rossfauldsbjss) [inline src/functions/pollJournals/framework/handler.ts]: process.env. variables need to be retrieved more securely in the future. (encrypted or pulled form secure vault)

File: package.json (hunk 1 above)
Comment (johnmorleybjss) [reply package.json]: Done

File: serverless.yml (hunk 2 above)
Comment (johnmorleybjss) [reply serverless.yml]: Done.

File: src/functions/pollJournals/framework/handler.ts (hunk 3 above)
Comment (johnmorleybjss) [reply src/functions/pollJournals/framework/handler.ts]: Added a config facade + Secrets Manager support.

**Pull request merged**